src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -399,9 +399,26 @@ class CharacterPrediction implements IPrediction {
 		return cursor.clone().moveTo(this.appliedAt.pos) + input;
 	}
 
-	public matches(input: StringReader, lookBehind?: IPrediction) {
+	public matches(input: StringReader, lookBehind?: IPrediction, cursor?: Cursor) {
 		let startIndex = input.index;
 
+		const cursorMove = input.eatRe(/^\x1b\[(\d+);(\d+)H/);
+		if (cursorMove) {
+			const x = this.appliedAt!.pos.x;
+			// We need to follow the cursor movements
+			cursor!.moveTo({
+				x: parseInt(cursorMove[2]) - 1,
+				y: parseInt(cursorMove[1]) - 1,
+				baseY: cursor!.baseY
+			});
+			while (cursor!.x < x) {
+				while (input.eatRe(CSI_STYLE_RE)) { }
```

Comment:
Might reuse the prediction omit loop here

Reply:
why?

Reply:
The style-eating logic is the same

Reply:
one gets added to the "output" but style doesn't. What is correct?